cli/src/cli.rs

```diff
@@ -1402,26 +1425,30 @@ pub fn process_command(config: &CliConfig) -> ProcessResult {
         CliCommand::AuthorizeNonceAccount {
             nonce_account,
             nonce_authority,
+            memo,
             new_authority,
         } => process_authorize_nonce_account(
             &rpc_client,
             config,
             nonce_account,
             *nonce_authority,
+            memo.as_ref(),
```

Comment:
```suggestion
            *memo,
```
With the `with_memo()` type change, can just take ownership of all of these.